com.playeveryware.eos/Runtime/Core/DLLHandle.cs

```diff
@@ -22,252 +22,203 @@
+        }
+
+        //-------------------------------------------------------------------------
+        public static string GetProductVersionForLibrary(string libraryName)
+        {
+            FileVersionInfo info = GetLibraryVersionInfo(libraryName);
 
-        foreach (string pluginPath in pluginPaths)
+            return info?.ProductVersion;
+        }
+
+        private static FileVersionInfo GetLibraryVersionInfo(string libraryName)
         {
-            foreach (var filesystemEntry in Directory.EnumerateFileSystemEntries(pluginPath, libraryName + ext, SearchOption.AllDirectories))
+            string libraryPath = GetPathForLibrary(libraryName);
+
+            return null == libraryPath ? null : FileVersionInfo.GetVersionInfo(libraryPath);
+        }
+
+        //-------------------------------------------------------------------------
+        public static string GetPathForLibrary(string libraryName)
+        {
+            List<string> pluginPaths = GetPathsToPlugins();
+
+            string extension = EOSManagerPlatformSpecificsSingleton.Instance != null
+                ? EOSManagerPlatformSpecificsSingleton.Instance.GetDynamicLibraryExtension()
+                : ".dll";
+
+            foreach (string pluginPath in pluginPaths)
             {
-                FileVersionInfo info = FileVersionInfo.GetVersionInfo(filesystemEntry);
-                print("Found : " + filesystemEntry);
-                if (info != null)
+                foreach (string entry in FileSystemUtility.GetFileSystemEntries(pluginPath, libraryName + extension))
                 {
-                    return string.Format(info.ProductVersion);
+                    return entry;
                 }
             }
-        }
-        return null;
-    }
 
-    //-------------------------------------------------------------------------
-    public static string GetPathForLibrary(string libraryName)
-    {
-        List<string> pluginPaths = GetPathsToPlugins();
-        string ext = ".dll";
+            return null;
+        }
 
-        foreach (string pluginPath in pluginPaths)
+        //-------------------------------------------------------------------------
+        public static DLLHandle LoadDynamicLibrary(string libraryName)
         {
-            foreach (var filesystemEntry in Directory.EnumerateFileSystemEntries(pluginPath, libraryName + ext, SearchOption.AllDirectories))
+            print("Loading Library " + libraryName);
+
+            string libraryPath = GetPathForLibrary(libraryName);
+
+            if (null == libraryPath)
             {
-                return filesystemEntry;
+                return null;
             }
+
+            IntPtr libraryHandle = SystemDynamicLibrary.Instance.LoadLibraryAtPath(libraryPath);
+            print("Trying to load with entry " + libraryPath);
+
+            return libraryHandle != IntPtr.Zero ? new DLLHandle(libraryHandle) : null;
```

Comment:
This change to remove the enumeration now won't throw an exception when it is unable to load a found dynamic library. In the previous code, it returns null if there is no matching dynamic library on disk, however if it finds a match on disk but `SystemDynamicLibrary.LoadLibraryAtPath` is unable to load it (i.e. it returns null), then it will throw as that is seen as an unrecoverable error.

The new code will return null in both cases.

Reply:
Good catch. I'm resolving that now.

Reply:
I have resolved this issue, consequently I am marking this as resolved.